### DISCUSSION TITLE
Solcast Forecast API count

### DISCUSSION THREAD

**Discussion by FozzieUK** [nathanmarlor/foxess_em#142] [Ideas]
I've just realised when setting up a new system that Solcast have recently reduced the number of API hits per day on new free accounts from 50 down to 10 - if you already have a 'legacy' account you continue to get the 50 but I guess it's only a matter of time before they think about reducing that.

As a new feature request is there any way of reducing the number of Solcast hits the api does - maybe a setting to choose high(50) / low(10) resolution, or perhaps get info via the alternative Forecast.solar integration.

Thanks

**Comment by nathanmarlor** [comment]
Ah, that's sad news! I wish they'd open up a 'premium' type subscription for home users, I'd be more than happy to pay for detailed, regular forecast info.

As it happens I have a change sat on a branch which gets the current number of API requests made from Solcast, then dynamically creates a refresh schedule. So if you have 50 at the start of the day, it'll refresh 50 times (or 25 if you have two sites). If you've restarted HA a few times and used a few API calls, or even if you've fired a few Solcast calls off manually, it'll pick up the reduced number of available API calls and recreate the schedule. 

I've been snowed recently and not had any time to spend on this, but I'll hopefully be able to grab some time soon to carry on making improvements :)

**Reply by FozzieUK** [reply]
Yes I would too, they seem geared around commercial clients and looking at the prices they charge I can see why - with a bit of luck they might introduce a low cost home user option - I'll email them and see what they say ;)

No problem with the timing, thanks for all of your work !

**Comment by nathanmarlor** [comment]
Solved in #155
